doc/mgmt/SONiC_overlay_ConfigDB.md

```diff
@@ -0,0 +1,338 @@
+      - When delete a static config item, profile underlay will add the item key to PROFILE_DELETE table.
+      - When read static config, any key in PROFILE_DELETE will not exist in result.
+      - When user set deleted item back, the item key will be remove from PROFILE_DELETE table.
+      - For example:
+        - BUFFER_POOL table are static config:
+    ```
+    "BUFFER_POOL": {
+        "ingress_lossless_pool": {
+            "size": "26531072",
+            "type": "ingress",
+            "mode": "dynamic",
+            "xoff": "6291456"
+        },
+        "egress_lossless_pool": {
+            "size": "32822528",
+            "type": "egress",
+            "mode": "static"
+        }
+    }
+  ```
+        - On mellanox devices, when set buffer mode to 'dynamic' mode, buffer manager will use swsscommon API to delete buffer pool to reclam unused buffer resource.
+        - When delete a static config item 'ingress_lossless_pool', the key of this item will be write to PROFILE_DELETE table. and swsscommon read API will not return this item anymore.
+        - Revert operation: when user set buffer mode back, buffer manager will write the deleted item back, when this happen, the key of the item will be remove from PROFILE_DELETE table. and swsscommon read API will return this item.
+
+# 2 Design
+
+- Underlay config DB design diagram:
+
+<img src="./images/swss-common-layer.png"  />
+
+- API design diagram:
+
+<img src="./images/swss-common-default-value.png"  />
+
+## 2.1 Considerations
+
+### Current design:
+
+- Existing read API keeps no change. 
+- Add decorator API to return default value and static config data.
+- Load static config as lazy as possible.
+
+### How to get default value
+
+|                                                               | Pros                                              | Cons                                                                                                                                              |
+| ------------------------------------------------------------- | ------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Get default value from Yang model in read API.                | Redis config DB keeps no change.                  | 3 MB memory per-process because need load Yang model and reference libyang.<br>50ms to load yang model.<br>8ms to read default value 10000 times. |
+| Write default value to default value DB when write config DB. | Better read performance, Less memory consumption. | Need add new Redis DB for default value.                                                                                                          |
+
+### How to get static config
+
+- Static config will stored in a new redis database 'PROFILE_DB', database index is 15.
```

Comment:
>15 [](http://example.com/codeflow?start=84&length=2)

Some UT is using 15 as TEST_DB. Please be aware for any conflict risk.

Reply:
Will test this issue with this PR: https://github.com/sonic-net/sonic-swss-common/pull/625
Will update here after all UT passed.